pkg/gpt-tools/files/zboot

```diff
@@ -0,0 +1,154 @@
+#!/bin/sh
+ZX="Zenix"
+
+set -e
+
+#
+# How do you get the root dev from a container? With the next two
+# functions.
+#
+
+makewords() {
+    awk '{print $0}' RS=' ' < $1
+}
+
+rdev () {
+    uuid=$(makewords /proc/cmdline | grep PARTUUID | awk -F = '{print $3}')
+    dev=$(lsblk -l -o name,PARTUUID | awk -v uuid="$uuid" '$2==uuid {print $1}')
+    echo $dev
+}
+
+rootdev=$(echo "/dev/"$(rdev))
+rootdsk=$(echo "/dev/"$(lsblk -no pkname ${rootdev}))
+
+usage() {
+    echo "${ZX} Boot Control and Administration API tool."
+    echo
+    echo "Usage:"
+    echo "       $(basename "$0") {partdev|partstate} <LABEL>"
+    echo "       $(basename "$0") set_partstate <LABEL> <STATE>"
+    echo "       $(basename "$0") {curpart|reset|watchdog}"
+    echo
+    echo "Commands:"
+    echo "    partdev: return device of partion LABEL."
+    echo "    partstate: return state of partition LABEL."
+    echo "    set_partstate: Set state of partition LABEL to STATE."
+    echo "    curpart: return label of current root partition."
+    echo "    watchdog: signal ${ZX} that the system is alive."
+    echo "    reset: reboot the machine with a certain hurry."
+    echo
+    echo "<IMAGE> can be IMGA, IMG, or any other GPT partition label."
```

Comment:
Does IMAGE refer to what is called LABEL above?

Reply:
Wops!